lib/mapbox/geocode.py

```diff
@@ -171,6 +171,20 @@ def get_latlng(address, min_relevance_threshold=1):
     return latlng
 
 
+def extract_address_info(response):
+    """Extract address, city, and country_code from a Mapbox feature."""
+    result = {}
+    result['address'] = response[0].get('place_name')
+    for ctx in response[0].get('context', []):
+        ctx_id = ctx.get('id', '')
+        if ctx_id.startswith('place.'):
+            result['city'] = ctx.get('text')
```

Comment:
Assuming that it is a city is presumptuous. Different countries/places will have different hierarchies.